app/controllers/users/sessions_controller.rb

```diff
@@ -1,7 +1,10 @@
 class Users::SessionsController < Devise::SessionsController
-  after_action :verify_authorized, except: [:new, :create, :destroy]
+  after_action :verify_authorized, except: [:new, :create, :destroy, :logout]
 
   def after_sign_in_path_for(resource)
     resource.after_sign_in_path
   end
+
+  def logout
```

Comment:
devise implements its own logout action and view. you might look into that first.